src/west/config.py

```diff
@@ -0,0 +1,56 @@
+'''
+Configuration file handling, using the standard configparser module.
+'''
+
+import configparser
+import os
+import platform
+
+from west.util import west_dir
+
+
+def read_config():
+    '''
+    Reads all configuration files, making the configuration values available as
+    a configparser.ConfigParser object in config.config. This object works
+    similarly to a dictionary: config.config['foo']['bar'] gets the value for
+    key 'bar' in section 'foo'.
+    '''
+
+    global config
+
+    #
+    # Gather (potential) configuration file paths, roughly following Git
+    # conventions.
+    #
+    # See the FILES section in git-config(1).
+    #
+
+    # System-wide
+
+    if platform.system() == 'Linux':
+        files = ['/etc/westconfig']
+    elif platform.system() == 'Darwin':  # Mac OS
+        # This was seen on a local machine ($(prefix) = /usr/local)
+        files = ['/usr/local/etc/westconfig']
+    elif platform.system() == 'Windows':
+        # Seen on a local machine. Not sure how it got derived though.
+        files = [os.path.expandvars('%PROGRAMDATA%\\west\\config')]
+
+    # Probably doesn't hurt to check $XDG_CONFIG_HOME and ~/.config/west/config
+    # on all systems. They're listed in git-config(1).
```

Comment:
I really don't like that but I don't think it should block the merge. My guess is Git works that way because it's a Linux program that happens to run on other platforms. The XDG_CONFIG_HOME and friends variables are part of the XDG base directory specification (https://standards.freedesktop.org/basedir-spec/basedir-spec-latest.html) and should only apply on those platforms in my opinion.

It will also make the documentation more awkward (and surprising IMO) for macOS and Windows users.

Reply:
We can live this out of the Windows/macOS doc in my opinion.

Reply:
I couldn't see anything about it being Linux-specific in that doc, but I might've missed it. If someone wants to set it on e.g. Mac OS, then more power to them, IMO.

I suspect the reason they do it is because it's nice, and take [Linus' stance on standards](https://lkml.org/lkml/2018/6/5/769). :D

Reply:
It's not nice, and neither that email from Linus

Reply:
I suspect Linus wouldn't write  that email today after his change of stance regarding several topics recently.